packages/model/src/event.ts

```diff
@@ -88,15 +89,61 @@ export type EventUpdateDataObject = Partial<
 export type EventCreateRequest = EventCreateDataObject;
 export type EventUpdateRequest = EventUpdateDataObject;
 
-type FilterOptions = {
-  workingGroupId?: string;
-  groupId?: string;
-  userId?: string;
-  externalAuthorId?: string;
-  teamId?: string;
-  googleId?: string;
-  hidden?: boolean;
-};
+type FilterOptions =
+  | {
+      workingGroupId?: string;
```

Comment:
might want to consider an intermediary type with all properties marked as `never` and then in each of the unions just override those which are to be given, but that's up to you. It's a little messy for that many properties being mutually exclusive but oh well...

Reply:
nice! I've refactored it here https://github.com/yldio/asap-hub/pull/3016/commits/15553e5069779762413fef4cede1a9657925ac45